src/api/install.cpp

```diff
@@ -0,0 +1,560 @@
+// Copyright (c) 2019, QuantStack and Mamba Contributors
+//
+// Distributed under the terms of the BSD 3-Clause License.
+//
+// The full license is in the file LICENSE, distributed with this software.
+
+#include "mamba/api/install.hpp"
+
+#include "mamba/core/channel.hpp"
+#include "mamba/api/configuration.hpp"
+#include "mamba/core/link.hpp"
+#include "mamba/core/mamba_fs.hpp"
+#include "mamba/core/output.hpp"
+#include "mamba/core/package_cache.hpp"
+#include "mamba/core/pinning.hpp"
+#include "mamba/core/subdirdata.hpp"
+#include "mamba/core/thread_utils.hpp"
+#include "mamba/core/transaction.hpp"
+#include "mamba/core/virtual_packages.hpp"
+
+#include "thirdparty/termcolor.hpp"
+
+
+namespace mamba
+{
+    void install()
+    {
+        auto& config = Configuration::instance();
+
+        config.load(MAMBA_ALLOW_ROOT_PREFIX | MAMBA_ALLOW_FALLBACK_PREFIX
+                    | MAMBA_ALLOW_EXISTING_PREFIX);
+
+        auto& install_specs = config.at("specs").value<std::vector<std::string>>();
+        auto& use_explicit = config.at("explicit_install").value<bool>();
+
+        if (!install_specs.empty())
+        {
+            if (use_explicit)
+            {
+                detail::install_explicit_specs(install_specs);
```

Comment:
personally, I'd prefer these two functions to be (at least) part of the API, if not the only API.

Reply:
I guess when adding a low-level API we should move it there because it only relies on `Context` configuration, which is not the case of high-level API that need the `Configuration` to be loaded to work correctly.

I can move it outside this namespace detail.

Reply:
no it's fine, keep it in that namespace. we can move that to core later.